x/liquidgov/keeper/msg_server_submit_tx.go

```diff
@@ -0,0 +1,50 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/Stride-Labs/stride/v5/utils"
+
+	epochstypes "github.com/Stride-Labs/stride/v5/x/epochs/types"
+	icqtypes "github.com/Stride-Labs/stride/v5/x/interchainquery/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/Stride-Labs/stride/v5/x/liquidgov/types"
+)
+
+func (k Keeper) MirrorProposals(ctx sdk.Context, hostZone stakeibctypes.HostZone) error {
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "Submitting ICQ for proposals to %s", hostZone.ChainId))
+
+	endTime := ctx.BlockTime().AddDate(1, 0, 0) // TODO veryify range query works, then change logic to compare to latest proposal on stride
```

Comment:
Naive question, what does endTime do? Is it querying all proposals that end before endTime, that end after endTime, or something else?

Reply:
This is outdated. I was attempting to pull a range based on the endTime but what it actually does is get an entry in the ActiveProposalQueue which only stores proposal id's. The proposal iterator uses it to get the end of the queue based on an endTime.